### PR TITLE
Touching record when restored.

### PR DESCRIPTION
**Pull request by leonardoalifraco**
According to what was discussed on #264 I've added the corresponding touch when the record is restored.

### CODE REVIEW

**Comment by abitdodgy** [comment]
@leonardoalifraco I see you've tried to take care of issue #264. Thanks for taking care of that. An aside, it's a good idea to check if someone has started work on something they have take ownership of before taking it on.

I'm curious, isn't there a better way than to use touch? When restoring we are already updating the `deleted_at` column, so why not take this opportunity and update the `updated_at` too instead of using `touch`, which, from what I understand, issues a separate query?

File: test/paranoia_test.rb [hunk 1]
```diff
@@ -757,6 +758,19 @@ def test_validates_uniqueness_still_works_on_non_deleted_records
...
   end
 
+  def test_updated_at_modification_on_restore
+    parent1 = ParentModel.create
+    pt1 = ParanoidModelWithTimestamp.create(:parent_model => parent1)
+    ParanoidModelWithTimestamp.record_timestamps = false
+    pt1.update_columns(created_at: 20.years.ago, updated_at: 10.years.ago, deleted_at: 10.years.ago) 
+    ParanoidModelWithTimestamp.record_timestamps = true
+    assert pt1.updated_at < 10.minutes.ago
+    assert !pt1.deleted_at.nil?
```
Comment (radar) [inline test/paranoia_test.rb]: I would use a `refute` here. I think that would work.

**Comment by radar** [comment]
LGTM besides that one comment. Let me know if the `refute` works or not and then I'll merge.

**Comment by lalifraco-devspark** [comment]
@abitdodgy Sorry to takeover this issue. But I was  investigating this gem and decided to start contributing with something that was already confirmed and asked for a change.
Regarding to the implementation details, it's making 2 different database queries, but the AR touch implementation encapsulates the modification of the updated_at/on column functionality and from my perspective the clearer path to address this would be to keep it DRY rather than copying the code.

@radar Great. I agree that the refute is clearer. I'll make the change.

**Comment by leonardoalifraco** [comment]
@radar I've already changed the assertion strategy using refute.

**Comment by radar** [comment]
Thanks @leonardoalifraco. I've been busy :)

**Pull request merged**